test-e2e/orbitTokenBridge.ts

```diff
@@ -0,0 +1,346 @@
+    deployerL2Wallet = new ethers.Wallet(deployerKey, l2Provider)
+
+    console.log('setupOrbitTokenBridge')
+    const { l1Network, l2Network } = await setupOrbitTokenBridge(
+      deployerL1Wallet,
+      deployerL2Wallet,
+      config.ethUrl,
+      config.arbUrl
+    )
+
+    _l1Network = l1Network
+    _l2Network = l2Network
+
+    // create user wallets and fund it
+    const userKey = ethers.utils.sha256(ethers.utils.toUtf8Bytes('user_wallet'))
+    userL1Wallet = new ethers.Wallet(userKey, l1Provider)
+    userL2Wallet = new ethers.Wallet(userKey, l2Provider)
+    await (
+      await deployerL1Wallet.sendTransaction({
+        to: userL1Wallet.address,
+        value: ethers.utils.parseEther('10.0'),
+      })
+    ).wait()
+  })
+
+  it('should have deployed token bridge contracts', async function () {
+    // get router as entry point
+    const l1Router = L1OrbitGatewayRouter__factory.connect(
+      _l2Network.tokenBridge.l1GatewayRouter,
+      l1Provider
+    )
+
+    expect((await l1Router.defaultGateway()).toLowerCase()).to.be.eq(
+      _l2Network.tokenBridge.l1ERC20Gateway.toLowerCase()
+    )
+    // expect((await l1Router.counterpartGateway()).toLowerCase()).to.be.eq(
+    //   _l2Network.tokenBridge.l2ERC20Gateway.toLowerCase()
+    // )
+  })
+
+  it('can deposit token via default gateway', async function () {
+    // fund user to be able to pay retryable fees
+    nativeToken = ERC20__factory.connect(_l2Network.nativeToken, userL1Wallet)
+    await (
+      await nativeToken
+        .connect(deployerL1Wallet)
+        .transfer(userL1Wallet.address, ethers.utils.parseEther('1000'))
+    ).wait()
+
+    // create token to be bridged
+    const tokenFactory = await new TestERC20__factory(userL1Wallet).deploy()
+    token = await tokenFactory.deployed()
+    await (await token.mint()).wait()
+
+    // snapshot state before
+
+    const userTokenBalanceBefore = await token.balanceOf(userL1Wallet.address)
+    const gatewayTokenBalanceBefore = await token.balanceOf(
+      _l2Network.tokenBridge.l1ERC20Gateway
+    )
+    const userNativeTokenBalanceBefore = await nativeToken.balanceOf(
+      userL1Wallet.address
+    )
+    const bridgeNativeTokenBalanceBefore = await nativeToken.balanceOf(
+      _l2Network.ethBridge.bridge
+    )
+
+    // approve token
+    const depositAmount = 350
+    await (
+      await token.approve(_l2Network.tokenBridge.l1ERC20Gateway, depositAmount)
+    ).wait()
+
+    // calculate retryable params
+    const maxSubmissionCost = 0
+    const callhook = '0x'
+
+    const gateway = L1OrbitERC20Gateway__factory.connect(
+      _l2Network.tokenBridge.l1ERC20Gateway,
+      userL1Wallet
+    )
+    const outboundCalldata = await gateway.getOutboundCalldata(
+      token.address,
+      userL1Wallet.address,
+      userL2Wallet.address,
+      depositAmount,
+      callhook
+    )
+
+    const l1ToL2MessageGasEstimate = new L1ToL2MessageGasEstimator(l2Provider)
```

Comment:
Are we doing all this manually because we dont have support in the sdk? If so, it would be nice to also e2e tests that uses the sdk methods after they're written. I guess this might live in the arb-sdk tests if it works there, but it might be easier just to have them here since the env is already set up here.

Reply:
Exactly, I'll switch to using SDK once it's done. Current implementation is more like sanity check that all's good and can be used as reference for SDK implementation